### PR TITLE
fix(nuxi): install nuxt as main dependency when doing nuxi upgrade

### PR DESCRIPTION
**Pull request by wokalek**
### 🔗 Linked issue

#22036 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Basically, projects based on the Nuxt framework depend on Nuxt directly and it is used as a core package. Accordingly, in cases when you work with Nuxt in production, you install dependencies via `pnpm install --prod`, while the Nuxt is not installed and you get an error.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by danielroe** [comment]
Currently we [deliberately add `nuxt` as a devDependency](https://github.com/nuxt/starter/blob/v3/package.json#L11-L15) but I think we could revisit this. Let's start with an issue to discuss: https://github.com/nuxt/nuxt/issues/22050

**Review by danielroe — CHANGES_REQUESTED** [review]
For now, can we respect whatever option the user has selected? ie. if nuxt is in their devDeps it will use `-D` flag and otherwise not?

It may be worth making this updated PR directly to https://github.com/nuxt/cli though, rather than here.

**Comment by pi0** [comment]
I think we might introduce this feature (and smart switch) for unjs/nypm util as well which is used in next CLI as new `upgradeDependency` utility.
